streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -400,65 +422,88 @@ public void closeDirty() {
      *  1. first close topology to make sure all cached records in the topology are processed
      *  2. then flush the state, send any left changelog records
      *  3. then flush the record collector
-     *  4. then commit the record collector -- for EOS this is the synchronization barrier
-     *  5. then checkpoint the state manager -- even if we crash before this step, EOS is still guaranteed
-     *  6. then if we are closing on EOS and dirty, wipe out the state store directory
-     *  7. finally release the state manager lock
      * </pre>
      *
-     * @param clean    shut down cleanly (ie, incl. flush and commit) if {@code true} --
+     * @param clean    shut down cleanly (ie, incl. flush) if {@code true} --
      *                 otherwise, just close open resources
-     * @throws TaskMigratedException if committing offsets failed (non-EOS)
-     *                               or if the task producer got fenced (EOS)
+     * @throws TaskMigratedException if the task producer got fenced (EOS)
      */
-    private void close(final boolean clean) {
+    private Map<TopicPartition, Long> prepareClose(final boolean clean) {
```

Comment:
`For the next PR`: I see the reason I return the checkpoint is that we are now extracting the committing out of the task and I need to remember if we need to checkpoint and if yes which offsets after we've flushed and before we checkpoint, but since the state of the task would not change before / after the commit during close. 

More specifically we only have three cases: 1) to not write checkpoint, 2) write checkpoints for written offsets (changelogs) only, 3) write checkpoint for written and consumed offsets. And no matter which case it is during the `preClose`, it would always be the same in the `post`, so why do we need to return it to task-manager, book-keep there, and then after commit to pass it back to tasks?

Reply:
By `next PR`, you mean the one after we finish the EOS-beta commit feature right?

Reply:
I mean the next PR when we add the EOS-beta feature --- please see the first comment I have with this tag.